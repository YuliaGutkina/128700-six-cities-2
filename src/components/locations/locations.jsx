import React from 'react';
import PropTypes from "prop-types";
import classNames from "classnames";
import {connect} from "react-redux";

import {receiveCitiesListSelector} from "../../reducer/data/selectors";
import {ActionCreator} from "../../reducer/data/data";


const Locations = (props) => {
  const {citiesList, currentCity, onTabClick} = props;

  return <div className="tabs">
    <section className="locations container">
      <ul className="locations__list tabs__list">
        {citiesList.map((city) => <li key={city.name} className="locations__item">
          <a
            className={
              classNames(
                  `locations__item-link`,
                  `tabs__item`,
                  {"tabs__item--active": (city.name === currentCity)}
              )}
            href="#"
            onClick={() => {
              onTabClick(city.name);
            }}
          >
            <span>{city.name}</span>
          </a>
        </li>)}
      </ul>
    </section>
  </div>;
};

Locations.propTypes = {
  currentCity: PropTypes.string,
  citiesList: PropTypes.array,
  onTabClick: PropTypes.func
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  citiesList: receiveCitiesListSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  onTabClick: (city) => {
    dispatch(ActionCreator.changeCity(city));
  },
});


export {Locations};
export default connect(mapStateToProps, mapDispatchToProps)(Locations);
