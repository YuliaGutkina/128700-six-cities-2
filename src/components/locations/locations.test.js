import React from "react";
import renderer from "react-test-renderer";

import {Locations} from "./locations";


it(`Offers list correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <Locations
          currentCity="Paris"
          citiesList={[]}
          onTabClick={jest.fn()}
        />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
