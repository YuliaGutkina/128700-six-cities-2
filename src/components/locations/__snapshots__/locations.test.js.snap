// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers list correctly renders after relaunch 1`] = `
<div
  className="tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    />
  </section>
</div>
`;
